### PR TITLE
doc:add more description about application constraints

### PR DESCRIPTION
**Pull request by wuxyintel**
In the current design guidlines, it is not clear about
how to check application constraints and how to record
them.

Add description about it is optional to do error checking
for application constraints during hypversisor boot time;
Add reference about how to record design assumptions.

Signed-off-by: Xiangyang Wu <xiangyang.wu@linux.intel.com>

### CODE REVIEW

File: doc/developer-guides/sw_design_guidelines.rst [hunk 1]
```diff
@@ -64,7 +64,9 @@ below:
   All application constraints shall be guaranteed by external safety
   applications, such as Board Support Package, firmware, safety VM, or Hardware.
   The verification of application integration shall check whether the safety
-  application meets all application constraints.
+  application meets all application constraints, hyppervisor tester should verify
```
Comment (dbkinder) [inline doc/developer-guides/sw_design_guidelines.rst]:
```suggestion
  application meets all application constraints. These constraints must be verified
```

File: doc/developer-guides/sw_design_guidelines.rst [hunk 2]
```diff
@@ -64,7 +64,9 @@ below:
...
+  it during validation test. It is optional to do error checking for application
```
Comment (dbkinder) [inline doc/developer-guides/sw_design_guidelines.rst]:
```suggestion
  during validation test. It is optional to do error checking for application
```

File: doc/developer-guides/sw_design_guidelines.rst [hunk 3]
```diff
@@ -64,7 +64,9 @@ below:
...
+  constraint at hypervisor boot time.
```
Comment (dbkinder) [inline doc/developer-guides/sw_design_guidelines.rst]:
```suggestion
  constraints at hypervisor boot time.
```

File: doc/developer-guides/sw_design_guidelines.rst [hunk 4]
```diff
@@ -73,6 +75,12 @@ below:
 
   - Input data which is only specified by external safety application.
 
+.. note:: If input data can be specified by both of non-safety VM and safety VM,
```
Comment (dbkinder) [inline doc/developer-guides/sw_design_guidelines.rst]:
```suggestion
.. note:: If input data can be specified by both a non-safety VM and a safety VM,
```

File: doc/developer-guides/sw_design_guidelines.rst [hunk 5]
```diff
@@ -73,6 +75,12 @@ below:
...
+   application constraint isn't applicable to these data, related error checking
```
Comment (dbkinder) [inline doc/developer-guides/sw_design_guidelines.rst]:
```suggestion
   the application constraint isn't applicable to these data.  Related error checking
```

File: doc/developer-guides/sw_design_guidelines.rst (hunk 1 above)
Comment (wuxyintel) [reply doc/developer-guides/sw_design_guidelines.rst]: Thanks for your comment and review, will update it.

File: doc/developer-guides/sw_design_guidelines.rst (hunk 2 above)
Comment (wuxyintel) [reply doc/developer-guides/sw_design_guidelines.rst]: Thanks for your comment and review, will update it.

File: doc/developer-guides/sw_design_guidelines.rst (hunk 3 above)
Comment (wuxyintel) [reply doc/developer-guides/sw_design_guidelines.rst]: Thanks for your comment and review, will update it.

File: doc/developer-guides/sw_design_guidelines.rst (hunk 4 above)
Comment (wuxyintel) [reply doc/developer-guides/sw_design_guidelines.rst]: Thanks for your comment and review, will update it.

File: doc/developer-guides/sw_design_guidelines.rst (hunk 5 above)
Comment (wuxyintel) [reply doc/developer-guides/sw_design_guidelines.rst]: Thanks for your comment and review, will update it.

**Pull request merged**